packages/inertia-react/src/useForm.js

```diff
@@ -16,7 +16,7 @@ export default function useForm(...args) {
   const [progress, setProgress] = useState(null)
   const [wasSuccessful, setWasSuccessful] = useState(false)
   const [recentlySuccessful, setRecentlySuccessful] = useState(false)
-  let transform = (data) => data
+  const transform = useRef((data) => data)
```

Comment:
Isn't it clearer to use the `useCallback` function? That way you can omit the `.current` parts, which will increase readability. 
https://reactjs.org/docs/hooks-reference.html#usecallback